### ISSUE TITLE
Download speed monitoring drops to zero when total reaches ~4 GiB (Linux ARM)

### ISSUE THREAD

**Issue by hiatsu0** [aristocratos/btop#224]
**Describe the bug**

Network download speed monitoring drops to 0 Byte/s after the Total reaches value 3.96 - 3.99 GiB on Raspberry Pi 4B (4GB RAM) running latest, up-to date Raspberry OS Lite (Bullseye)

**To Reproduce**

Start btop, wait for transfer amount reach approx. 4 GiB => Speed indication drops to 0.
Quit btop, restart and the counter will run to 4 GiB again, showing the speed until then.

**Screenshots**
<img width="885" alt="0 byte" src="https://user-images.githubusercontent.com/66969014/147159819-18e29610-36cc-4fd5-854f-74485f5f569a.png">

**Info (please complete the following information):**
 - btop++ version: 1.1.3
 - Binary: static
 - Architecture: armv7l
 - Platform: Linux (Raspberry OS / Bullseye)
 - (Linux) Kernel: 5.10.63-v7l+
 - Terminal used: bash

**Additional context**

contents of `~/.config/btop/btop.log`

```
2021/12/23 (00:03:18) | ===> btop++ v.1.1.3
2021/12/23 (00:03:18) | DEBUG: Starting in DEBUG mode!
2021/12/23 (00:03:18) | INFO: Logger set to DEBUG
2021/12/23 (00:03:18) | DEBUG: Using locale en_GB.UTF-8
2021/12/23 (00:03:18) | INFO: Running on /dev/pts/0
2021/12/23 (00:03:18) | INFO: Forcing tty mode: setting 16 color mode and using tty friendly graph symbols
2021/12/23 (00:05:27) | DEBUG: Writing new config file
2021/12/23 (00:05:27) | INFO: Quitting! Runtime: 00:02:09
```

**Comment by cookieMr** [comment]
I have the same issue on MacOS.

**Info:**
 - btop version: 1.1.3
 - Architecture: x86_64
 - Platform: OsX
 - (Linux) Kernel: 20.6.0
 - (OSX/FreeBSD) Os release version: 11.6 (macOS Big Sur)
 - Terminal used: Terminal 2.11 (440)
 - Font used: default

<img width="745" alt="Screen Shot 2021-12-29 at 11 29 12 AM" src="https://user-images.githubusercontent.com/8577083/147653483-ac4dd207-4b53-4591-9891-4490db1a2c4f.png">

**Comment by aristocratos** [comment]
@cookieMr 
Well that is very odd since I thought this would have been a 32-bit issue.

@cookieMr and @hiatsu0 
If you press `z` and zero the totals is there any difference in the speed value, also what are you using for comparing the speed values?

**Comment by cookieMr** [comment]
@aristocratos 
Nope. No difference. I hit the `z` key and the download statistics are showing 0 all over the places.
Let me mention here, that i still continue to download a huge file when i hit the `z` key and that download (actual & non-zero) speed is not shown on the stats.

<img width="751" alt="Screen Shot 2021-12-29 at 3 36 29 PM" src="https://user-images.githubusercontent.com/8577083/147673281-b6f655a6-770a-40d4-8ebf-344daa310a61.png">

**Comment by userdocs** [comment]
@hiatsu0 according to this https://www.raspberrypi.com/products/raspberry-pi-4-model-b/specifications/ 

You have this processor `Broadcom BCM2711, Quad core Cortex-A72 (ARM v8) 64-bit SoC @`

And the OS has arm64 images

https://downloads.raspberrypi.org/raspios_arm64/images/

Can you confirm you are actually using the 32 (x86) version of the OS since you said your arch is:

Architecture: armv7l

There is no reason to no be using the arm64 image if the hardware info is correct.

**Comment by hiatsu0** [comment]
@userdocs: That is correct. I just happened to install the 32 bit version because (at least at the time) the RPi imager app did not offer the 64 bit and I could not be bothered to download it manually as my use case is so simple there’s no difference btw 32 & 64. And I think the 64 bit version is still not officially supported/recommended by the foundation?

@aristocratos: I don’t currently have access to the RPi 4 I noticed this bug on but I can try with my older RPi 3B running the same OS version. I can confirm what @cookieMr is seeing though. I did the testing moving large files to an SMB share, and only way to see the transfer speed after ~4GiB total was to restart btop, none of the buttons could reset the situation. If I recall correct, usually the total sum would show ~1GiB already at the btop restart, sometimes ~3GiB, so I got a feeling this would some sort of ”rolling counter” not in btop but in the system’s services — but I have no idea.

**Comment by aristocratos** [comment]
@hiatsu0 @cookieMr 
I've added code to account for the system rolling over the network values for both linux and macos.
The internal rollover in btop won't happen until the total usage is over 9223 Petabytes(!), but I've added code to reset to 0 if that happens, future proofing... :)

Try the binaries from the [Continuous Builds](https://github.com/aristocratos/btop/actions) or remake with latest commit.

**Comment by hiatsu0** [comment]
@aristocratos: Can confirm the bug on 1.1.4 on RPi 3B (`5.10.63-v7+ armv7l`) and also not fixed in Continuous Build [`56deec4`](https://github.com/aristocratos/btop/commit/56deec49885498be0cca99717c59107e8e88ff16)

Here's wget actively running and download speed went to 0 as soon as total reached ~4 GiB:
<img width="932" alt="bug2" src="https://user-images.githubusercontent.com/66969014/147782250-ef9a112b-ecea-45ab-a2f9-e6f6d90dbe6c.png">

**Comment by cookieMr** [comment]
Same for me.

I `brew uninstall btop` anf then `make && sudo make install` from the `main` branch. The speed drops to 0 at 3.99GB.

**Comment by aristocratos** [comment]
@cookieMr @hiatsu0 
Should be fixed in latest commit, thanks to @pixelbart80

**Comment by aristocratos** [comment]
Fixed in release 1.2.0
